components/team_settings_modal.jsx

```diff
@@ -27,108 +29,104 @@ class TeamSettingsModal extends React.Component {
     constructor(props) {
         super(props);
 
-        this.updateTab = this.updateTab.bind(this);
-        this.updateSection = this.updateSection.bind(this);
-
         this.state = {
             activeTab: 'general',
             activeSection: ''
         };
     }
-    componentDidMount() {
-        const modal = $(ReactDOM.findDOMNode(this.refs.modal));
-
-        modal.on('click', '.modal-back', function handleBackClick() {
-            $(this).closest('.modal-dialog').removeClass('display--content');
-            $(this).closest('.modal-dialog').find('.settings-table .nav li.active').removeClass('active');
-        });
-        modal.on('click', '.modal-header .close', () => {
-            setTimeout(() => {
-                $('.modal-dialog.display--content').removeClass('display--content');
-            }, 500);
-        });
 
+    componentDidMount() {
         if (!Utils.isMobile()) {
             $('.settings-modal .settings-content').perfectScrollbar();
         }
     }
-    updateTab(tab) {
-        this.setState({activeTab: tab, activeSection: ''});
+
+    updateTab = (tab) => {
+        this.setState({
+            activeTab: tab,
+            activeSection: ''
+        });
+
         if (!Utils.isMobile()) {
             $('.settings-modal .modal-body').scrollTop(0).perfectScrollbar('update');
         }
     }
-    updateSection(section) {
+
+    updateSection = (section) => {
         if ($('.section-max').length) {
             $('.settings-modal .modal-body').scrollTop(0).perfectScrollbar('update');
         }
+
         this.setState({activeSection: section});
     }
+
+    closeModal = () => {
+        this.props.onModalDismissed();
+    }
+
+    collapseModal = () => {
+        $(ReactDOM.findDOMNode(this.refs.modalBody)).closest('.modal-dialog').removeClass('display--content');
```

Comment:
Any thoughts on how we could remove this for something more React-friendly? I see that the UserSettingsModal does the same thing here, but it would be nice to get rid of this

Reply:
I'll look into it and see what I find.